hamcrest/Hamcrest/File/FileMatcher.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @author Vasek Brychta <vaclav@brychtovi.cz>
+ */
+
+namespace Hamcrest\File;
+
+use Hamcrest\BaseMatcher;
+use Hamcrest\Description;
+
+abstract class FileMatcher extends BaseMatcher
+{
+	/** @var string */
+	private $ownDescription;
+	/** @var string */
+	private $failureDescription;
+
+	/**
+	 * @param string $ownDescription
+	 * @param string $failureDescription
+	 */
+	public function __construct($ownDescription, $failureDescription)
+	{
+		$this->ownDescription = $ownDescription;
+		$this->failureDescription = $failureDescription;
+	}
+
+	public function describeTo(Description $description)
+	{
+		$description->appendText($this->ownDescription);
+	}
+
+	final public function matches($item)
+	{
+		return $this->isSafeType($item) && $this->matchesFile($this->createSplFileInfoObjectFromPath($item));
+	}
+
+	final public function describeMismatch($item, Description $mismatchDescription)
+	{
+		if ($this->isSafeType($item))
+		{
+			$this->describeFileMismatch($this->createSplFileInfoObjectFromPath($item), $mismatchDescription);
+		}
+		else
+		{
+			parent::describeMismatch($item, $mismatchDescription);
+		}
```

Comment:
Code formatting. Generally, it's the best idea to configure IDE as per the coding standards of the project and mass-reformat any new code.

Reply:
Where can I get the coding standards for this project?

Reply:
I have no idea. The https://github.com/hamcrest/hamcrest-php/blob/master/CONTRIBUTING.md file mentions coding standard but doesn't tell what it is. I'm assuming it's PSR-2.

Anyway, it's usually best to copy/paste new files from parts of the existing file to ensure that the code style is the same.